examples/platforms/simulation/radio.c

```diff
@@ -887,3 +950,51 @@ otError otPlatRadioGetCoexMetrics(otInstance *aInstance, otRadioCoexMetrics *aCo
     return error;
 }
 #endif
+
+uint64_t otPlatRadioGetNow(void)
+{
+    return otPlatTimeGet();
+}
+
+#if OPENTHREAD_CONFIG_CSL_RECEIVER_ENABLE
+static void updateIeData(otInstance *aInstance)
+
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    uint8_t offset = 0;
+
+#if OPENTHREAD_CONFIG_CSL_RECEIVER_ENABLE
+    if (sCslPeriod > 0)
+    {
+        memcpy(sAckIeData, sCslIeHeader, IE_HEADER_SIZE);
+        offset += IE_HEADER_SIZE + CSL_IE_SIZE; // reserve space for CSL IE
+    }
+#endif
+    if (sIeState.mLinkMetricsEnabled)
+    {
+        // TODO
+    }
+    sAckIeDataLength = offset;
+}
+#endif // OPENTHREAD_CONFIG_CSL_RECEIVER_ENABLE
+
+#if OPENTHREAD_CONFIG_CSL_RECEIVER_ENABLE
+otError otPlatRadioEnableCsl(otInstance *aInstance, uint32_t aCslPeriod, const otExtAddress *aExtAddr)
```

Comment:
Should simulation also record parent's extended address?

Reply:
I think it's unneccessary. We provided the parent's extended address on Nordic platform because its radio driver need it in order to generate auto-ack with CSL IE(It has a table which maintains what IEs should be included in the ack sent to different extended addresses). Our implementation on simulation to generate the enhanced ack don't need the extended address.